src/fidesapi/routes/generate.py

```diff
@@ -0,0 +1,153 @@
+"""
+Contains all of the endpoints required to manage generating resources.
+"""
+from enum import Enum
+from typing import Dict, List, Optional, Union
+
+from fastapi import APIRouter, HTTPException, Response, status
+from fideslang.models import Organization, System
+from loguru import logger as log
+from pydantic import BaseModel
+
+from fidesapi.routes.crud import get_resource
+from fidesapi.routes.util import API_PREFIX, unobscure_string
+from fidesapi.sql_models import sql_model_map
+from fidesctl.core.system import generate_aws_systems
+
+
+class GenerateTypes(str, Enum):
+    """
+    Generate Type Enum to capture the discrete possible values
+    for a valid type of resource to generate.
+    """
+
+    SYSTEM = "systems"
+    DATASET = "datasets"
+
+
+class AWSConfig(BaseModel):
+    """
+    The model for the connection config for AWS
+    """
+
+    region_name: str
+    aws_secret_access_key: str
+    aws_access_key_id: str
+
+
+class OktaConfig(BaseModel):
+    """
+    The model for the connection config for Okta
+    """
+
+    okta_client_token: str
+
+
+class Generate(BaseModel):
+    """
+    Defines attributes for generating resources included in a request.
+    """
+
+    config: Union[AWSConfig, OktaConfig]
+    target: str
+    type: GenerateTypes
+
+
+class GenerateRequestPayload(BaseModel):
+    """
+    The model for the request body housing generate information.
+    """
+
+    organization_key: str
+    generate: Generate
+
+
+class GeneratedResponse(BaseModel):
+    """
+    The model to hous the response for generated infrastructure.
+    """
+
+    generate_results: Optional[List[System]]
+
+
+router = APIRouter(tags=["Generate"], prefix=f"{API_PREFIX}/generate")
+
+
+@router.post(
+    "/",
+    status_code=status.HTTP_200_OK,
+    response_model=GeneratedResponse,
+    responses={
+        status.HTTP_400_BAD_REQUEST: {
+            "content": {
+                "application/json": {
+                    "example": {"detail": "Unable to generate resources."}
+                }
+            }
+        },
+    },
+)
+async def generate(
+    generate_request_payload: GenerateRequestPayload, response: Response
+) -> Dict:
+    """
+    A multi-purpose endpoint to generate Fides resources based on existing
+    infrastructure
+
+    Currently generates Fides resources for the following:
+    * AWS: Systems
+
+    In the future, this will include options for other Systems & Datasets,
+    examples include:
+    * Okta: Systems
+    * Snowflake: Datasets
+
+    All config secrets should be encoded as a minor security precaution, using the
+    `obscure_string` function in `fidesapi.routes.util`
+
+    All production deployments should implement HTTPS for security purposes
+    """
+    organization = await get_resource(
+        sql_model_map["organization"], generate_request_payload.organization_key
+    )
+    if generate_request_payload.generate.target.lower() == "aws":
```

Comment:
I would rather we return a 400 Bad Request if the `target` is not known. 

Should we also check the type here? target=aws type=datasets should not be allowed. Maybe we can offload the validation to another function that can raise a 400.

Reply:
I like this 👍🏽

Reply:
Checked and this returned a 500, but some extra validation should return a 422 if the `target` is not known with latest change